rfc/1216-resolve-cloud-tokens.md

```diff
@@ -0,0 +1,845 @@
+to fetch the runtime attributes of a resource. However, not all CloudFormation resources are currently supported by Cloud Control.
+
+> For a list of supported resources, see [here](https://docs.aws.amazon.com/cloudcontrolapi/latest/userguide/supported-resources.html).
+
+In case you need an attribute for such a resource, you can add a `CfnOutput` resource to your stack, and use the resolver to 
+explicitly fetch its value from the deployed stack. For example:
+
+```ts
+...
+
+// create the output
+const bucketNameOutput = new aws.CfnOutput(this, 'BucketName', {
+  value: bucket.bucketName,
+});
+
+const bucketName = resolver.fetchOutput(bucketNameOutput);
+
+new kplus.CronJob(manifest, 'CronJob', {
+  schedule: k8s.Cron.daily(),
+  containers: [{
+    image: 'job',
+    envVariables: {
+      // passing the output value
+      BUCKET_NAME: kplus.EnvValue.fromValue(bucketName),
+    }
+ }]
+});
+
+...
+```
+
+#### CDK For Terraform
+
+In this example, we create an S3 `Bucket` with the CDKTF, and pass its (deploy time generated) name 
+as an environment variable to a Kubernetes `CronJob` resource.
+
+```ts
+import * as tf from "cdktf";
+import * as aws from "@cdktf/provider-aws";
+import * as k8s from 'cdk8s';
+import * as kplus from 'cdk8s-plus-26';
+
+import { CdkTfTokenResolver } from '@cdk8s/cdktf-token-resolver';
+
+const awsApp = new tf.App();
+const stack = new tf.TerraformStack(awsApp, 'Stack');
+
+const resolver = new CdkTfTokenResolver(stack);
+
+const k8sApp = new k8s.App();
+const manifest = new k8s.Chart(k8sApp, 'Manifest', { resolver });
+
+const bucket = new aws.s3Bucket.S3Bucket(stack, 'Bucket');
+
+new kplus.CronJob(manifest, 'CronJob', {
+  schedule: k8s.Cron.daily(),
+  containers: [{
+    image: 'job',
+    envVariables: {
+      // passing the bucket name via an env variable
+      BUCKET_NAME: kplus.EnvValue.fromValue(bucket.bucket),
+    }
+ }]
+});
+
+awsApp.synth();
+k8sApp.synth();
+```
+
+Notice we create two applications: one for our cdk8s constructs, and one for our CDKTF constructs.
+Both are defined and synthesized in the same file, but can be separated as needed.
+Since your Kubernetes resources now depend on CDKTF tokens, you'll first need to run `cdktf deploy`, 
+and only then `cdk8s synth`.
+
+> Otherwise, the Kubernetes manifests will contain a string representation of the tokens (e.g `${TfToken[TOKEN.0]}`), 
+> instead of the concrete values. To learn more CDKTF tokens, see [here](https://developer.hashicorp.com/terraform/cdktf/concepts/tokens).
+
+---
+
+Ticking the box below indicates that the public API of this RFC has been
+signed-off by the API bar raiser (the `api-approved` label was applied to the
+RFC pull request):
+
+```
+[ ] Signed-off by API Bar Raiser @xxxxx
+```
+
+## Public FAQ
+
+### What are we launching today?
+
+This launch consists of three deliverables:
+
+- A new feature in the `cdk8s` core library that allows injecting custom resolution logic during synthesis.
```

Comment:
Given that now all 3 CDKs have tokens or token-related features... wouldn't it make sense to put down some motivation why this feature shouldn't be added to `constructs` by now? 

Yes, it would probably become a bigger project. But wouldn't it be better for everyone to rethink this infrastructure from the ground up, including all the features we're now stitching on to it?

Reply:
As we discussed, this feature is not introducing a new token system into cdk8s, it merely allows interpreting tokens of other systems. And since we are using `CfnOutput` instead of Cloud Control, we don't even need to replicate the handling of CloudFormation intrinsics, so I don't see anything thats worth pulling out to `constructs`.